readthedocs/proxito/tests/test_old_redirects.py

```diff
@@ -533,40 +531,62 @@ def test_not_found_page_without_trailing_slash(self):
             )
 
 
-# FIXME: these tests are valid, but the problem I'm facing is that the request
-# is received as ``GET '//my.host/path/'`` (note that we are losing the http:)
-@pytest.mark.xfail(strict=True)
+@override_settings(
+    PYTHON_MEDIA=True,
+    PUBLIC_DOMAIN="dev.readthedocs.io",
+    ROOT_URLCONF="readthedocs.proxito.tests.handler_404_urls",
+)
 class UserRedirectCrossdomainTest(BaseDocServing):
 
     def test_redirect_prefix_crossdomain(self):
-        """
-        Avoid redirecting to an external site unless the external site is in to_url
-        """
+        """Avoid redirecting to an external site unless the external site is in to_url."""
         fixture.get(
             Redirect,
-            project=self.project, redirect_type='prefix',
-            from_url='/',
-        )
-
-        r = self.client.get(
-            'http://testserver/http://my.host/path.html',
-            HTTP_HOST='project.dev.readthedocs.io',
-        )
-        self.assertEqual(r.status_code, 302)
-        self.assertEqual(
-            r['Location'],
-            'http://project.dev.readthedocs.io/en/latest/http://my.host/path.html',
-        )
-
-        r = self.client.get(
-            'http://testserver//my.host/path.html',
-            HTTP_HOST='project.dev.readthedocs.io',
-        )
-        self.assertEqual(r.status_code, 302)
-        self.assertEqual(
-            r['Location'],
-            'http://project.dev.readthedocs.io/en/latest/my.host/path.html',
-        )
+            project=self.project,
+            redirect_type="prefix",
+            from_url="/",
+        )
+
+        urls = [
+            # Plain protocol, these are caught by the slash redirect.
+            (
+                "http://project.dev.readthedocs.io/http://my.host/path.html",
+                "/http:/my.host/path.html",
```

Comment:
```suggestion
                "http://project.dev.readthedocs.io/http://my.host/path.html",
                "/http://my.host/path.html",
```

I think there is a small typo here.

Reply:
this test is correct, we are being redirected by the "slash redirect" that's in our middleware, basically it removes duplicates.

https://github.com/readthedocs/readthedocs.org/blob/f08b94be3b858efd9a7053d9b3b46436e3526163/readthedocs/proxito/middleware.py#L286-L287

Reply:
Would be good to have a comment in that test case, below the URL, explainig this.